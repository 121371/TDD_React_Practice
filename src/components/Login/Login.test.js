import React from "react";
import { shallow } from "enzyme";

import Form from "./Login";

describe("Form component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Form />);
  });

  it("component renders correctly", () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it("should render an username input tag", () => {
    expect(wrapper.find('input[name="username"]').exists()).toBe(true);
  });

  it("username must have the default valaue", () => {
    expect(wrapper.find('input[name="username"]').props().value).toBe('');
  });

  it("should render a password input tag", () => {
    expect(wrapper.find('input[name="password"]').exists()).toBe(true);
  });

  it("password must have the default valaue", () => {
    expect(wrapper.find('input[name="password"]').props().value).toBe('');
  });

  it("should render a submit button", () => {
    expect(wrapper.find('button[name="login-button"]').exists()).toBe(true);
  });

  
});
