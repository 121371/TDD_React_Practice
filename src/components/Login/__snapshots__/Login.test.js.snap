// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component component renders correctly 1`] = `
Array [
  <div>
    <input
      name="username"
      onChange={[Function]}
      type="text"
    />
    <input
      name="password"
      onChange={[Function]}
      type="password"
    />
    <button
      name="login-button"
      onClick={[Function]}
    >
      Login
    </button>
  </div>,
]
`;
